src/axom/quest/SignedDistance.hpp

```diff
@@ -608,34 +676,22 @@ inline double SignedDistance<NDIMS, ExecSpace>::computeSign(
   const MinCandidate& currMin)
 {
   double sgn = 1.0;
-  // STEP 1: Select the pseudo-normal N at the closest point to calculate the
-  // sign.
+  // STEP 1: Select the pseudo-normal N at the closest point to calculate the sign.
   // There are effectively 3 cases based on the location of the closest point.
   VectorType N;
-  if(currMin.minLoc >= TriangleType::NUM_TRI_VERTS)
+  switch(currMin.minType)
   {
+  case detail::ClosestPointLocType::face:
     // CASE 1: closest point is on the face of the surface element
     N = currMin.minTri.normal();
-  }
-  else if(currMin.minLoc < 0)
-  {
-    // CASE 2: closest point is on an edge, use sum of normals of equidistant
-    // faces
-    // TODO: Sometimes, the traversal fails to find the opposite face, so only
-    // a single face's normal is accumulated here. The proper solution would be
-    // to precompute edge pseudo-normals during construction time, but that
-    // would also require generating cell-to-face connectivity for the surface
-    // mesh.
+    break;
+  default:  // Use precomputed normal for edges and vertices
```

Comment:
I remember when debugging the original SignedDistance PR that the BVH traversal sometimes only accumulated one face of an edge pseudo-normal; was that the underlying issue here?

In any case, I think it might be good to have a debug check/tracker variable here for the edge and vertex pseudo-normals:
- If we keep track of the number of normals we accumulated for the current weighted sum, an edge closest point should have exactly two accumulated normals, one for each face.
- Not sure about the angle-weighted normal for vertex closest points, but I think we could check that there at least 3? (since any angle at a triangle vertex should be less than 180 degrees)

Reply:
> I remember when debugging the original SignedDistance PR that the BVH traversal sometimes only accumulated one face of an edge pseudo-normal; was that the underlying issue here?

There were a few problems. Most prominently:
1.  `primal::closest_point()` was using strict inequality tests, so points that were within `EPS of an edge or vertex could be marked as interior (face), and either not counted for the edge/vertex test, or would clear out the normal
2. The logic for determining whether to clear normals and/or sum normals needed some improvement.

Reply:
The underlying issue is that we're using purely geometric tests, so we cannot guarantee that the we're catching all (both) faces in the star of an edge or all faces in the star of a vertex.

A potentially better, but more expensive way to handle this would be to:
1. Weld the mesh to ensure that we have a mesh instead of a collection of disjoint faces (see #705)
2. Compute the pseudo-normal at each vertex of the welded mesh
3. Only keep track of the single closest point
4. Derive the sign from the normal of the mesh at this point. 

This would robustly handle the face, edge and vertex cases for a watertight surface mesh.
The other downside of this approach would be that it would involve more memory (to store the normals) and more computation time (to weld the mesh and compute the normals)

Reply:
We can certainly add a debug check/variable to count the number of accumulated normals.
What would you suggest we do with that information though? 
I suppose we can add an option to add scalar fields to the output vtk mesh to visualize this (closest point type and count)